server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -2112,11 +2112,13 @@ private boolean doCancelMaintenance(final long hostId) {
 
         /* TODO: move to listener */
         _haMgr.cancelScheduledMigrations(host);
+
+        boolean vms_migrating = false;
         final List<VMInstanceVO> vms = _haMgr.findTakenMigrationWork();
         for (final VMInstanceVO vm : vms) {
-            if (vm != null && vm.getHostId() != null && vm.getHostId() == hostId) {
-                s_logger.info("Unable to cancel migration because the vm is being migrated: " + vm);
-                return false;
+            if (vm.getHostId() != null && vm.getHostId() == hostId) {
```

Comment:
Is this conditional right?
`vm.getHostId() != null && vm.getHostId() == hostId`

It is looking a little weird to me.